react-components/src/data-providers/core-dm-provider/getCadConnectionsForRevisions.ts

```diff
@@ -0,0 +1,194 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { type QueryRequest } from '@cognite/sdk/dist/src';
+import { type FdmCadConnection, type FdmKey } from '../../components/CacheProvider/types';
+import { type NodeItem, type DmsUniqueIdentifier, type FdmSDK } from '../FdmSDK';
+import {
+  COGNITE_3D_OBJECT_SOURCE,
+  COGNITE_ASSET_SOURCE,
+  COGNITE_CAD_NODE_SOURCE,
+  type CogniteAssetProperties,
+  type CogniteCADNodeProperties,
+  CORE_DM_3D_CONTAINER_SPACE,
+  CORE_DM_SPACE
+} from './dataModels';
+import { cogniteCadNodeSourceWithProperties } from './cogniteCadNodeSourceWithProperties';
+import { getModelIdFromExternalId, getRevisionIdFromExternalId } from './getCdfIdFromExternalId';
+import { createFdmKey } from '../../components/CacheProvider/idAndKeyTranslation';
+import { type PromiseType } from '../utils/typeUtils';
+import { isDefined } from '../../utilities/isDefined';
+import { type QueryResult } from '../utils/queryNodesAndEdges';
+import { restrictToDmsId } from './restrictToDmsId';
+import { cogniteAssetSourceWithProperties } from './cogniteAssetSourceWithProperties';
+
+export async function getCadConnectionsForRevisions(
+  modelRevisions: Array<[DmsUniqueIdentifier, DmsUniqueIdentifier]>,
+  fdmSdk: FdmSDK
+): Promise<FdmCadConnection[]> {
+  const results = await getModelConnectionResults(modelRevisions, fdmSdk);
+  const object3dToAssetMap = createObject3dToAssetMap(results.items.assets);
+  const cadNodeToModelMap = createCadNodeToObject3dMap(results.items.cad_nodes);
+
+  const returnResult = results.items.cad_nodes
+    .map((cadNode) => {
+      const props = cadNode.properties[CORE_DM_SPACE]['CogniteCADNode/v1'];
+      const object3dKey = cadNodeToModelMap.get(createFdmKey(cadNode));
+
+      if (object3dKey === undefined) {
+        return undefined;
+      }
+
+      const connectedAsset = object3dToAssetMap.get(object3dKey);
+
+      if (connectedAsset === undefined) {
+        return undefined;
+      }
+
+      const modelIdRevisionIdTreeIndex = getModelAndTreeIndex(modelRevisions, props);
+
+      if (modelIdRevisionIdTreeIndex === undefined) {
+        return undefined;
+      }
+
+      return {
+        ...modelIdRevisionIdTreeIndex,
+        instance: restrictToDmsId(connectedAsset)
+      };
+    })
+    .filter(isDefined);
+
+  return returnResult;
+}
+
+function createObject3dToAssetMap<T extends NodeItem<CogniteAssetProperties>>(
+  assets: T[]
+): Map<FdmKey, T> {
+  return new Map(
+    assets.map((asset) => [
+      createFdmKey(asset.properties[CORE_DM_SPACE]?.['CogniteAsset/v1'].object3D),
```

Comment:
Does make sense to move the key string 'CogniteAsset/v1' into a const?

Reply:
I did it :+1: